src/services/messagesService.ts

```diff
@@ -130,22 +130,21 @@ export default class MessagesService {
   }
 
   private parseResponse<T>(
-    res: IResponseType<number, T> | undefined
+    res: IResponseType<number, Error | T> | undefined
   ): Either<ServiceError, T> {
     // If the response is undefined (can't be decoded) or the status is not 200 dispatch a failure action.
     if (!res) {
       log.error(logErrorOnDecodeError, res);
       return left<ServiceError, T>(internalError(messageErrorOnApiError));
     }
-
     if (res.status === 200) {
-      return right<ServiceError, T>(res.value);
-    } else if (res.status === 404) {
+      return right<ServiceError, T>(res.value as T);
```

Comment:
we have to cast here. another option is to pass the T type as a parameter to this function then use io-ts decode here.
I don't know the api client internals (ie. if it already calls decode before returning the response) so I've chose to avoid a double decoding here. cc @cloudify

Reply:
yes the client is supposed to decode the response to the io-ts type